src/main/java/marquez/service/JobService.java

```diff
@@ -242,13 +260,52 @@ public Job createOrUpdate(
     }
   }
 
+  private void updateRunInputDatasets(
+      UUID runUuid,
+      List<DatasetRow> inputRows,
+      NamespaceName namespaceName,
+      JobName jobName,
+      JobVersionRow jobVersion) {
+    if (inputRows.isEmpty()) {
+      return;
+    }
+    List<RunInput> runInputs = new ArrayList<>();
+    for (DatasetRow inputDataset : inputRows) {
+      Optional<DatasetVersionRow> version =
+          datasetVersionDao.findLatestDatasetByUuid(inputDataset.getUuid());
+      if (version.isPresent()) {
+        DatasetVersionRow row = version.get();
+        runDao.updateInputVersions(runUuid, version.get().getUuid());
+        runInputs.add(
+            new RunInput(
+                new DatasetVersionId(
+                    namespaceName, DatasetName.of(inputDataset.getName()), row.getUuid())));
+      }
+    }
+    Optional<ExtendedRunRow> runRow = runDao.findBy(runUuid);
+    if (runRow.isEmpty()) {
+      return;
+    }
+    ExtendedRunRow run = runRow.get();
+    notify(
+        new JobInputUpdate(
+            RunId.of(runUuid),
+            new RunMeta(
+                RunId.of(runUuid),
+                run.getNominalStartTime().orElse(null),
+                run.getNominalEndTime().orElse(null),
+                ImmutableMap.of()),
```

Comment:
what was this last parameter for?

Reply:
Arguments of the run. I feel like its unnecessary so I would lean more towards removing RunMeta completely rather than reconstructing the object with dao calls.

Reply:
I think we'd want to return the run args as well. The **class** [`ExtendedRunRow`](https://github.com/MarquezProject/marquez/blob/main/src/main/java/marquez/db/models/ExtendedRunRow.java) returns the run args as a Json string (key / value pairs) and can be deserialized using `Utils.fromJson()`. For example https://github.com/MarquezProject/marquez/blob/main/src/main/java/marquez/service/mappers/Mapper.java#L382

We'd might want to add a `toRunArgs()` method to the mapper that factors our the  deserialization logic.